x/execution/internal/keeper/keeper.go

```diff
@@ -170,3 +249,60 @@ func (k *Keeper) List(ctx sdk.Context) ([]*executionpb.Execution, error) {
 	iter.Close()
 	return execs, nil
 }
+
+// parseAccAddresses parses multiple accAddresses at once.
+func parseAccAddresses(addresses []*ownershippb.Ownership) ([]sdk.AccAddress, error) {
+	var out []sdk.AccAddress
+	for i := range addresses {
+		// send to first one for now
+		addr, err := sdk.AccAddressFromBech32(addresses[i].Owner)
+		if err != nil {
+			return nil, err
+		}
+		out = append(out, addr)
+	}
+	return out, nil
+}
+
+// coinsQuoRem divides coins and returns quotient and remainder.
+// NOTE: if reminder is zero then it will be set to quotient.
+func coinsQuoRem(coins sdk.Coins, div int) (sdk.Coins, sdk.Coins) {
```

Comment:
can't we use directly the arithmetic function on the type `coins` or `coin`?

Reply:
I spot `DecCoins` yesterday as well, but I know the api of `math/big` and It was easier for me to write, because DecCoins lacks docs... 😠 But I can change it to deccoins today.

Reply:
Please try with de coins ;) the go doc is your api :p